stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/MarkReadImplTest.kt

```diff
@@ -1,31 +1,94 @@
+    }
+
+    private lateinit var chatDomain: ChatDomainImpl
+    private lateinit var client: ChatClient
+
+    private lateinit var sut: MarkReadImpl
+
+    @BeforeEach
+    fun init() {
+        client = mock()
+        chatDomain = mock {
+            on { scope } doReturn testCoroutines.scope
+        }
+        sut = MarkReadImpl(chatDomain)
+    }
+
+    @Test
+    fun `Given empty cid Should throw exception`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            sut.invoke("")
+        }
+    }
 
     @Test
-    fun read() = testCoroutines.dispatcher.runBlockingTest {
-        val channelControllerImpl = chatDomainImpl.channel(data.channel1.cid)
-        channelControllerImpl.handleEvent(data.newMessageFromUser2)
-        advanceUntilIdle()
-        var unreadCount = channelControllerImpl.unreadCount.getOrAwaitValue()
-        Truth.assertThat(unreadCount).isEqualTo(1)
-
-        val result = chatDomain.useCases.markRead(data.channel1.cid).execute()
-        advanceUntilIdle()
-        assertSuccess(result)
-        val lastRead = channelControllerImpl.read.getOrAwaitValue()?.lastRead
-        Truth.assertThat(lastRead).isEqualTo(data.messageFromUser2.createdAt)
-        unreadCount = channelControllerImpl.unreadCount.getOrAwaitValue()
-        Truth.assertThat(unreadCount).isEqualTo(0)
-        Truth.assertThat(channelControllerImpl.toChannel().unreadCount).isEqualTo(0)
+    fun `Given invalid cid Should throw exception`() {
+        assertThrows(IllegalArgumentException::class.java) {
+            sut.invoke("no-colon")
+        }
+    }
+
+    @Test
+    fun `Given valid cid Should invoke markRead() on ChannelController`() = runBlockingTest {
+        val channelController = mock<ChannelControllerImpl>()
+        val cid = randomCID()
+        whenever(chatDomain.channel(cid)) doReturn channelController
+        sut.invoke(cid).execute()
+        verify(channelController).markRead()
+    }
+
+    @Test
+    fun `Given valid cid and successful ChannelController result Should return call client markRead() and return Result(true)`() = runBlockingTest {
+        val channelController = mock<ChannelControllerImpl> {
+            onBlocking { markRead() } doReturn true
+        }
+        val channelType = "messaging"
+        val channelId = randomString()
+        val cid = randomCID()
+        whenever(chatDomain.channel(cid)) doReturn channelController
+        whenever(chatDomain.client) doReturn client
+        whenever(channelController.channelType) doReturn channelType
+        whenever(channelController.channelId) doReturn channelId
+        val markReadCall = mock<Call<Unit>>()
+        whenever(client.markRead(any(), any())) doReturn markReadCall
+        val markReadCallResult = mock<Result<Unit>>()
+        whenever(markReadCall.execute()) doReturn markReadCallResult
+
+        val result = sut.invoke(cid).execute()
+
+        verify(channelController).markRead()
```

Comment:
IMO in this test, it's better to split it for two. Usually, they are two different kinds of tests.
1) `State tests` - when you check the state of some objects (checking fields to be equal to some values)
2) `Behavioral tests` - when you check that something happened (mockito's `verify`)

Reply:
done